### ISSUE TITLE
Unable to upgrade from Mint 21.3 to Mint 22

### ISSUE THREAD

**Issue by Mr-Green-Genes** [linuxmint/mintupgrade#104]
I have a Starlabs Byte II (small desktop device, not portable) which came preloaded with Mint 21.3.  I started the process of upgrading to Mint 22 but hit a brick wall.  I had a message which said "Connect the computer to a power source ..." and wouldn't go any further.  Using **acpi -a** returns **Adapter 0: off-line**, **acpi -b** gives **Battery 0: Discharging, 0%** and **hostnamectl** chassis returns **desktop.**

What is going on here please and how can I fix it?

**Comment by Jeremy7701** [comment]
mintupgrade runs `acpi -a` and a return value containing "off-line" causes the error.

However, a Starlabs Byte II claims it comes with a 36W power adaptor - which (if true) would act like a battery charger on a laptop?

**Comment by Mr-Green-Genes** [comment]
Yes, it does have a power adapter like a laptop.  So is **acpi -a** returning "off line" a BIOS issue?

**Comment by GPard1009** [comment]
Hi, I have the same problem. I am using a mini desktop too. Same error message. "Connect the computer to a power source ..."

hostnamectl shows Chassis: desktop
acpi -a returns Adapter 0: off-line
acpi -b returns nothing

 I have bypassed the mintupgrade check by setting the 'check.result' in the mintmupgrade.py to 'RESULT_SUCCESS'  and the upgrade worked, BUT it installed Mint22 like a laptop and the power management showed that the computer is running on battery, which does not exist.

I am running on Mint21.3 Kernel: Linux 5.15.0-117-generic I have upgraded to Kernel 6.8.039. The result was that the computer did not power off after shutting down Mint.

I hope someone has an idea what to do.
Does it make sense to open another bug on the same issue ?

cheers G. Pard

**Comment by Jeremy7701** [comment]
Definitely worth opening as an issue if it consistently doesn't power off.

**Comment by Jeremy7701** [comment]
Does entering `sudo shutdown now` in a terminal do anything?

**Comment by Mr-Green-Genes** [comment]
I think this should be directed to #105.  I do not have that issue as I am still on Mint 21.3 and Kernel 6.5.0-45-generic and can shut down ok.

**Comment by GPard1009** [comment]
have tried sudo shutdown -P and the result was the same. Linux does shut down but it does not power off.

I have tried it a several times also with sudo shutdown now

With 5.15.0-118-generic there is no problem, the computer does power off

Here something for the update problem. I have installed 21.3 on the same computer in a VM (Virtual Box) and mintupgrade does not find a connected powersupply. 

Updating this VM to 6.8.0.39 shows no problem to 'power off' ok. its not really the same thing.

**Comment by Jeremy7701** [comment]
The problem is that mintupgrade runs 'acpi -a' and checks that the output does not include the string "off-line".
Since its essential that laptops are plugged in, it seems unlikely that this will be disabled.

I'd suggest one of two possible workrounds if you want to edit a system program:-
**Obviously I haven't tried testing these.**
You might like to try this out in a VM first.

1. Locate where mintupgrade runs 'acpi -a' and change it to run 'acpi -v'
An appropriate location appears to be line 184 of `/usr/lib/linuxmint/mintupgrade/checks.py`
Alternatively...
2. Compile any C "hello, world" program as `/usr/bin/acpi` and do not change mintupgrade.
Reinstall acpi afterwards, if you should take this course.
